How do I get an owned Info?

Cargo.toml:

```toml
[package]
name = "repro"
version = "0.1.0"
edition = "2021"

[dependencies]
png = "0.17.11"
```

main.rs:

```rust
use std::fs::File;
use std::path::Path;

use png::{Decoder, DecodingError, Info};

fn main() {
    decode_and_return_info("irrelevant.png").unwrap();
}

fn decode_and_return_info<P: AsRef<Path>>(png: P) -> Result<Info<'static>, DecodingError> {
    let dec = Decoder::new(File::open(png.as_ref()).unwrap());
    let mut rdr = dec.read_info()?;
    rdr.finish()?;
    Ok(rdr.info().clone().to_owned())
}
```

Expected result: Code compiles.

Actual result:

```
error[E0515]: cannot return value referencing local variable `rdr`
  --> src\main.rs:14:5
   |
14 |     Ok(rdr.info().clone().to_owned())
   |     ^^^---^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |     |  |
   |     |  `rdr` is borrowed here
   |     returns a value referencing data owned by the current function

error: aborting due to previous error
```

I'm trying to read a PNG, make slight modifications, then save a new PNG, and ran into the above error when trying to put the decoding part into its own function. I can't find any way to get an `Info<'static>`.

--- a/src/decoder/mod.rs
+++ b/src/decoder/mod.rs
@@ -202,7 +202,7 @@ impl<R: Read> Decoder<R> {
     ///
     /// Most image metadata will not be read until `read_info` is called, so those fields will be
     /// None or empty.
-    pub fn read_header_info(&mut self) -> Result<&Info, DecodingError> {
+    pub fn read_header_info(&mut self) -> Result<&Info<'static>, DecodingError> {
         let mut buf = Vec::new();
         while self.read_decoder.info().is_none() {
             buf.clear();
@@ -345,7 +345,7 @@ impl<R: Read> ReadDecoder<R> {
         ))
     }
 
-    fn info(&self) -> Option<&Info> {
+    fn info(&self) -> Option<&Info<'static>> {
         self.decoder.info.as_ref()
     }
 }
@@ -466,7 +466,7 @@ impl<R: Read> Reader<R> {
     /// Get information on the image.
     ///
     /// The structure will change as new frames of an animated image are decoded.
-    pub fn info(&self) -> &Info {
+    pub fn info(&self) -> &Info<'static> {
         self.decoder.info().unwrap()
     }
 

